### PR TITLE
Upgrade to TF 0.12 compliant syntax

### PR DESCRIPTION
**Pull request by jonandernovella**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This PR aims at upgrading the terraform config files to TF 0.12 compliant syntax.
### Changes made:

1) All modules have been upgraded to new syntax
2) Fixes have been made in order for terraform interpret list as string (flatten())

### Mentions:

@blankdots

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+ENV PLUGIN_RKE=0.12.1
+ENV PLUGIN_KUBERNETES=1.7.0
+ENV PLUGIN_NULL=2.1.2
+ENV PLUGIN_LOCAL=1.3.0
 ENV PLUGIN_TEMPLATE=2.0.0
-ENV PLUGIN_RANDOM=2.0.0
+ENV PLUGIN_RANDOM=2.1.2
 
 # PIP version
 ENV PIP=9.0.3
```
Comment (blankdots) [inline Dockerfile]: out of curiosity do we need such an old  PIP version ? current version is `19.1.1`

File: deployment-template/modules/node/variables.tf [hunk 2]
```diff
@@ -1,65 +1,65 @@
...
-variable labels {
-  type        = "map"
+variable "labels" {
+  type        = map(string)
   description = "Map of Kubernetes labels for this node"
   default     = {}
 }
 
-variable ssh_bastion_host {
-  default = "Bastion SSH host (mandatory if assign_floating_ip is false)"
```
Comment (blankdots) [inline deployment-template/modules/node/variables.tf]: don't remove the line but keep it as `"description"`

File: rega.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def version(image):
 @click.option('-B', '--backend', default='local',
               type=click.Choice(['local', 's3', 'swift']),
               help='Options are: "local", "s3" and "swift"')
-@click.option('-C', '--config', default="backend.config",
+@click.option('-C', '--config', default="backend.cfg",
```
Comment (blankdots) [inline rega.py]: in `Readme.md` this change from `.config` to `.cfg` should be reflected as well.

File: Dockerfile (hunk 1 above)
Comment (blankdots) [reply Dockerfile]: we upgrade it below, so nevermind

File: rega.py (hunk 3 above)
Comment (jonandernovella) [reply rega.py]: Done.

**Comment by jonandernovella** [comment]
> should we take into consideration the warning bellow?
> 
> ```
> Warning: the rke_node_parameter will be deprecated and removed in a future version. Use rke_cluster+dynamic instead
> 
> 
> 
>   on modules/node/main.tf line 40, in data "rke_node_parameter" "node_mappings":
> 
>   40: data "rke_node_parameter" "node_mappings" {
> ```

This variable will be deprecated in the future, so it's a good idea to replace it. I achieved this by using a new `null_data_source` in conjunction with the new Terraform 0.12 dynamic node block configuration syntax. That results in `nodes_conf` variable no longer being used in the `rke` module.

File: deployment-template/modules/node/variables.tf (hunk 2 above)
Comment (jonandernovella) [reply deployment-template/modules/node/variables.tf]: Done! Good catch!

**Pull request merged**